lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+          }
+          // Fill in apiMap.
+          let promises = [];
+          for (let i = 0; i < browserAPIs.length; i += 1) {
+            let interfaceId = browserAPIs[i].interfaceId;
+            promises.push(this.interfaces
+              .where(this.mlang.EQ(this.WebInterface.ID, interfaceId)).select()
+              .then((arrayDao) => {
+                let interfaceName = arrayDao.a[0].interfaceName;
+                let APIName = arrayDao.a[0].APIName;
+                if (!apiMap[interfaceName] ||
+                  typeof apiMap[interfaceName] !== 'object') {
+                  // If typeof is not object, this may be a build-in function.
+                  apiMap[interfaceName] = {};
+                }
+                if (!apiMap[interfaceName][APIName] ||
+                  typeof apiMap[interfaceName][APIName] !== 'object') {
+                  // If visit APIName for first time,
+                  // initialize an array of false.
+                  apiMap[interfaceName][APIName] = [];
+                  for (let k = 0; k < numBrowsers; k += 1) {
+                    apiMap[interfaceName][APIName].push(false);
+                  }
+                }
+                apiMap[interfaceName][APIName][
+                  browserDict[browserAPIs[i].browserId]] = true;
+              }));
+          }
+          Promise.all(promises).then(function() {
+            resolve(apiMap);
+          });
+        }.bind(this));
+      },
+    },
+    {
+      name: 'toMap',
+      documentation: `Takes an array of browser keys, produce a JSON containing
+        browser and interface relationship`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `Returns a JSON of the form:
+                {interfaceName: {APIName: [Boolean ...]} ...}`,
+      },
+      code: function(browserKeys) {
+        return new Promise(function(resolve) {
+          let browserIds = browserKeys.map((key) => this.getBrowserId_(key));
+          this.browserAPIs
+            .where(this.mlang.IN(this.BrowserAPI.BROWSER_ID, browserIds))
+            .select()
+            .then((arrayDao) => {
+              resolve(this.toMap_(arrayDao.a, browserKeys));
+            });
+        }.bind(this));
+      },
+    },
+    {
+      name: 'toCSV',
+      documentation: `Takes an array of browser keys, produce a String of
+        CSV format`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys`,
+        },
+      ],
+      returns: {
+        typeName: 'String',
+        documentation: `Returns a string of csv format`,
+      },
+      code: function(browserKeys) {
+        return new Promise(function(resolve) {
```

Comment:
Let's discuss this strategy. What I had in mind after our meeting was that procedures such as this would use something like:

```
// DAO query pseudo-code:
SELECT interface_id FROM apis ORDER BY interface_name
SELECT * FROM browser_apis ORDER BY browser_id GROUP BY interface_id
for each interface_id: use GROUP BY result to extract browser_ids that are True; fill in True/False in ordered-by-browser_id columns
```

Maybe there are results from experimenting with different approaches that led to this implementation. Let's chat offline.